lib/active_model_serializers/adapter/json_api/deserialization.rb

```diff
@@ -182,23 +182,29 @@ def parse_relationship(assoc_name, assoc_data, options)
           prefix_key = field_key(assoc_name, options).to_s.singularize
           hash =
             if assoc_data.is_a?(Array)
-              { "#{prefix_key}_ids".to_sym => assoc_data.map { |ri| ri['id'] } }
+              { "#{prefix_key}_ids".to_sym => assoc_data.map { |ri| ri[:id] } }
```

Comment:
Just FYI, this is likely going to change very soon. I've been altering the transform stuff to be more flexible. Stay tuned for that.